src/HDF5.jl

```diff
@@ -738,25 +738,12 @@ d_create(parent::Union{File,Group}, path::String, dtype::Type, dspace_dims; pv..
 
 # Note that H5Tcreate is very different; H5Tcommit is the analog of these others
 t_create(class_id, sz) = Datatype(h5t_create(class_id, sz))
-function t_commit(parent::Union{File,Group}, path::String, dtype::Datatype, lcpl::Properties, tcpl::Properties, tapl::Properties)
-    h5p_set_char_encoding(lcpl.id, cset(typeof(path)))
-    h5t_commit(checkvalid(parent).id, path, dtype.id, lcpl.id, tcpl.id, tapl.id)
+function t_commit(parent::Union{File,Group}, path::AbstractString, dtype::Datatype, lcpl::Properties=p_create(H5P_LINK_CREATE), tcpl::Properties=DEFAULT_PROPERTIES, tapl::Properties=DEFAULT_PROPERTIES)
```

Comment:
I'm going to split this change into the other PR, leave this PR to clean up imports.